src/models/converters/data_decoded.rs

```diff
@@ -73,3 +75,48 @@ impl DataDecoded {
         }
     }
 }
+
+impl DataDecoded {
+    pub(super) fn build_address_info_index(
+        &self,
+        info_provider: &mut impl InfoProvider,
+    ) -> Option<HashMap<String, AddressInfo>> {
+        let mut index = HashMap::new();
+        if self.method == MULTI_SEND {
+        } else {
+            if let Some(parameters) = &self.parameters {
+                for parameter in parameters {
+                    match &parameter.value {
+                        ParamValue::SingleValue(value) => {
+                            if value.len() == 42 && value.starts_with("0x") {
+                                if let Some(address_info) =
+                                    info_provider.full_address_info_search(&value).ok()
+                                {
+                                    index.insert(value.to_owned(), address_info);
+                                };
+                            }
+                        }
+                        ParamValue::ArrayValue(values) => {
+                            for value in values {
+                                if let ParamValue::SingleValue(value) = value {
+                                    if value.len() == 42 && value.starts_with("0x") {
```

Comment:
Could we extract these couple lines into a method to avoid code duplication?

Reply:
Planning on doing that as soon as I've also implemented the multiSend branch and covered with unit tests :)